File: src/features/invites/dto/accept-invite.dto.ts
import { IsString, Matches } from 'class-validator';
import { STRONG_PASSWORD_MESSAGE, STRONG_PASSWORD_REGEX } from '../../auth/dto/password-rules';

export class AcceptInviteDto {
  @IsString()
  token: string;

  @IsString()
  username: string;

  @IsString()
  @Matches(STRONG_PASSWORD_REGEX, { message: STRONG_PASSWORD_MESSAGE })
  password: string;
}

File: spring-boot-api/src/main/java/com/mytegroup/api/dto/invites/AcceptInviteDto.java
package com.mytegroup.api.dto.invites;

import com.mytegroup.api.dto.auth.PasswordRules;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AcceptInviteDto {
    
    @NotBlank(message = "Token is required")
    private String token;
    
    @NotBlank(message = "Username is required")
    private String username;
    
    @NotBlank(message = "Password is required")
    @Pattern(regexp = PasswordRules.STRONG_PASSWORD_REGEX, message = PasswordRules.STRONG_PASSWORD_MESSAGE)
    private String password;
    
    public void setUsername(String username) {
        this.username = username != null ? username.trim() : null;
    }
}
